tapfreighter/wallet.go

```diff
@@ -1654,6 +1662,15 @@ func addAnchorPsbtInputs(btcPkt *psbt.Packet, vPkt *tappsbt.VPacket,
 	lastIdx := len(btcPkt.UnsignedTx.TxOut) - 1
 	currentFee := inputAmt - outputAmt
 	feeDelta := int64(requiredFee) - currentFee
+	changeValue := btcPkt.UnsignedTx.TxOut[lastIdx].Value
+
+	// The fee may exceed the total value of the change output, which means
+	// this spend is impossible with the given inputs and fee rate.
+	if changeValue-feeDelta < 0 {
```

Comment:
I think we can just call `CheckTransactionSanity` again here? It should handle catching the negative output. 

Even with that though, if we ended up in that case, even with a fee estimator that returned "nothing" (so defaults to 253 sat/kw), we have a bug here if we're arriving at negative outputs.